### PR TITLE
feat: add `--request-interval` parameter

### PR DESCRIPTION
**Pull request by danielboloc**
## Overview

It adds a new parameter `--request-interval` to the `job run` command, allowing for a custom input of time to request job completion status to the API. An example would be:

```
cloudos job run ... --wait-completion --request-interval 5
```

The constant variable `REQUEST_INTERVAL` is removed, replaced by the `--request-interval`, which has a default value of `30`. The change was also done in `cromwell start` command, to be able to remove the constant.

Issue addressed in this PR:

#90 

Link to Sprint Issue:

[CLOUDOSCLI-4](https://lifebit.atlassian.net/browse/CLOUDOSCLI-4)

## Changes
- `CHANGELOG.md`
- `README.md`
- Add `--request-interval` option
- Bump version from 1.0.0 to 1.1.0

## Test
```
docker run --rm -it quay.io/lifebitaiorg/cloudos-cli:v1.1.0

MY_API_KEY="****"
CLOUDOS="https://cloudos.lifebit.ai"
WORKSPACE_ID="****"
PROJECT_NAME="API request interval"
WORKFLOW_NAME="rnatoy"

cloudos job run \
    --cloudos-url $CLOUDOS \
    --apikey $MY_API_KEY \
    --workspace-id $WORKSPACE_ID \
    --project-name "$PROJECT_NAME" \
    --workflow-name $WORKFLOW_NAME \
    --parameter reads=s3://lifebit-featured-datasets/pipelines/rnatoy-data \
    --parameter genome=s3://lifebit-featured-datasets/pipelines/rnatoy-data/ggal_1_48850000_49020000.Ggal71.500bpflank.fa \
    --parameter annot=s3://lifebit-featured-datasets/pipelines/rnatoy-data/ggal_1_48850000_49020000.bed.gff \
    --spot \
    --wait-completion \
    --request-interval 25
```

Expected output:

```
CloudOS python package: a package for interacting with CloudOS.

Version: 1.1.0

CloudOS job functionality: run and check jobs in CloudOS.

Executing run...
	Job successfully launched to CloudOS, please check the following link: https://cloudos.lifebit.ai/app/jobs/633590c43735bf0154ac048f
	Your assigned job id is: 633590c43735bf0154ac048f
	Please, wait until job completion or max wait time of 3600 seconds is reached.
	Your current job status is: initializing.
	Your current job status is: running.

	Your job took 450 seconds to complete successfully.
```

cloudOS job: https://cloudos.lifebit.ai/app/jobs/633590c43735bf0154ac048f

### CODE REVIEW

**Review by dapineyro — CHANGES_REQUESTED** [review]
Thanks @danielboloc ! I've added some comments and suggestions. In addition to them, please, remember to create the docker container for the corresponding version and push it to quay.io

File: cloudos/__main__.py [hunk 1]
```diff
@@ -496,11 +502,18 @@ def cromwell_status(cromwell_token,
 @click.option('--verbose',
               help='Whether to print information messages or not.',
               is_flag=True)
+@click.option('--request-interval',
+              help=('Time interval to request (in seconds) the job status. ' +
+                    'For large jobs is important to use a high number to ' +
+                    'make fewer requests so that is not considered spamming by the API. ' +
+                    'Default=30.'),
+              default=30)
```
[the remark this answers is not in the document]
Comment (danielboloc) [reply cloudos/__main__.py]: Removed this option from `cromwell`

File: cloudos/__main__.py [hunk 2]
```diff
@@ -184,8 +190,8 @@ def run(apikey,
             elapsed = 0
             while elapsed < 300 and c_status_h != 'Running':
                 c_status_old = c_status_h
-                time.sleep(REQUEST_INTERVAL)
-                elapsed += REQUEST_INTERVAL
+                time.sleep(request_interval)
+                elapsed += request_interval
```
[the remark this answers is not in the document]
Comment (danielboloc) [reply cloudos/__main__.py:194]: I agree, now for the cromwell there is the constant `REQUEST_INTERVAL_CROMWELL`, set to 30

[the remark this answers is not in the document]
Comment (dapineyro) [reply cloudos/__main__.py:194]: @danielboloc Please, change this `request_interval` here to be `REQUEST_INTERVAL_CROMWELL`, as this piece of code is in charge of the cromwell server restart.

**Review by dapineyro — CHANGES_REQUESTED** [review]
Thanks @danielboloc for the changes. Only make sure that all the `request_interval` are changed to `REQUEST_INTERVAL_CROMWELL` when interacting with cromwell server

**Review by dapineyro — APPROVED** [review]
LGTM!

**Pull request merged**